modules/auxiliary/gather/enum_dns.rb

```diff
@@ -1,483 +1,420 @@
       ], self.class)
 
     register_advanced_options(
       [
-        OptInt.new('RETRY', [ false, "Number of times to try to resolve a record if no response is received", 2]),
-        OptInt.new('RETRY_INTERVAL', [ false, "Number of seconds to wait before doing a retry", 2]),
-        OptBool.new('TCP_DNS', [false, "Run queries over TCP", false]),
+        OptInt.new('TIMEOUT', [false, 'DNS TIMEOUT', 8]),
+        OptInt.new('RETRY', [false, 'Number of times to try to resolve a record if no response is received', 2]),
+        OptInt.new('RETRY_INTERVAL', [false, 'Number of seconds to wait before doing a retry', 2]),
+        OptBool.new('TCP_DNS', [false, 'Run queries over TCP', false])
       ], self.class)
   end
 
-  def switchdns(target)
-    if not datastore['NS'].nil?
-      print_status("Using DNS Server: #{datastore['NS']}")
-      @res.nameserver=(datastore['NS'])
-      @nsinuse = datastore['NS']
-    else
-      querysoa = @res.query(target, "SOA")
-      if (querysoa)
-        (querysoa.answer.select { |i| i.class == Net::DNS::RR::SOA}).each do |rr|
-          query1soa = @res.search(rr.mname)
-          if (query1soa and query1soa.answer[0])
-            print_status("Setting DNS Server to #{target} NS: #{query1soa.answer[0].address}")
-            @res.nameserver=(query1soa.answer[0].address)
-            @nsinuse = query1soa.answer[0].address
-          end
-        end
-      end
+  def run
+    domain = datastore['DOMAIN']
+    return if domain.blank?
+
+    dns_wildcard(domain) unless datastore['STOP_WLDCRD']
+
+    if datastore['ENUM_STD']
+      get_a(domain)
+      get_cname(domain)
+      get_ns(domain)
+      get_mx(domain)
+      get_soa(domain)
+      get_txt(domain)
     end
+
+    get_tld(domain) if datastore['ENUM_TLD']
+    get_srv(domain) if datastore['ENUM_SRV']
+    axfr(domain) if datastore['ENUM_AXFR']
+    dns_bruteforce(domain) if datastore['ENUM_BRT']
   end
 
-  def wildcard(target)
-    rendsub = rand(10000).to_s
-    query = @res.query("#{rendsub}.#{target}", "A")
-    if query.answer.length != 0
-      print_status("This domain has wildcards enabled!!")
-      query.answer.each do |rr|
-        print_status("Wildcard IP for #{rendsub}.#{target} is: #{rr.address.to_s}") if rr.class != Net::DNS::RR::CNAME
+  def dns_query(domain, type)
+    begin
+      dns = Net::DNS::Resolver.new
+      nameserver = "#{datastore['NS']}"
+      unless nameserver.blank?
+        dns.nameservers -= dns.nameservers
+        dns.nameservers = ("#{datastore['NS']}")
       end
-      return true
-    else
-      return false
+      dns.use_tcp = datastore['TCP_DNS']
+      dns.udp_timeout = datastore['TIMEOUT']
+      dns.retry_number = datastore['RETRY']
+      dns.retry_interval = datastore['RETRY_INTERVAL']
+      dns.query(domain, type)
+    rescue ::Rex::ConnectionError
+    rescue ::Rex::ConnectionRefused
+    rescue ::Rex::ConnectionTimeout
+    rescue ::Rex::SocketError
+    rescue ::Rex::TimeoutError
```

Comment:
When would any of these get thrown?  I may be missing something, but it doesn't look like Rex is used at all any more for socket stuff.